pkg/api/api.go

```diff
@@ -224,3 +228,139 @@ type Message struct {
 	MessageID ulid.ULID   `json:"message_id"`
 	Payload   []byte      `json:"payload,omitempty"`
 }
+
+type Validator interface {
+	// checks whether this field is one of the valid options
+	Validate() error
+}
+
+// checks every field of the state, returning an error when any invalid field is encountered
+func (state State) Validate() error {
+
+	for _, field := range []Validator{
+		state.VisionStatus,
+		state.BallFound,
+		state.LocalizationStatus,
+		state.CPB,
+		state.Role,
+		state.RefBoxRole,
+		state.HomeGoal,
+		state.TeamColor,
+		state.CapacitorState,
+		state.Kinect1State,
+		state.Kinect2State,
+	} {
+		// the String() method is required, and also works for non-string Validators
+		if reflect.ValueOf(field).String() != reflect.Zero(reflect.TypeOf(field)).String() {
```

Comment:
Why do you call `String()`?

Reply:
without it, it doesn't work. 
Probably something to do with comparing pointers. Added it to the comment

Reply:
This should be fixed instead of introducing dirty hacks